### DISCUSSION TITLE
Is there an option to select LoRaWAN Version like 1.0.3?

### DISCUSSION THREAD

**Discussion by drucknase** [jgromes/RadioLib#916] [General]
it is the first time, I try to use RadioLib. Thank you for all the work on the library!
Up to now I always used MCCI LMIC 4.1.0 Lib for my nodes. The board I used for the trial is Lilygo TTGO-T-Motion. The board is build around the S76G ASIC from ACSIP. This ASIC includes an STM32L073RZ, a Sony CD5603 GPS-Module and a Semtech SX1276 LoRaWAN Chip.
On all my LoRaWAN Nodes I use LoRaWAN Version 1.0.3. From my understanding RadioLib uses as default LoRaWAN Version 1.1.0.
Question: can I configure Version 1.0.3?

Thank you for support

**Comment by jgromes** [comment]
RadioLib doesn't enforce any particular LoRaWAN version, whatever you set in your server/network provider is what will get used. For example, if you use TTN, you get to choose which version of LoRaWAN is used for the network. RadioLib supports 1.0.x and 1.1.

Though I will note that using 1.1 is highly recommended, seeing as 1.0 has significant security issues.

**Reply by drucknase** [reply]
Thank you for your very fast response! OK, so I will do some more experiments. From the LoRaWAN-node debug console it seems to work with success, but I see no join or other messages for the LoRaWAN-node on the TTN console.  I read some TTN troubleshooting and found some comments, that maybe the LoRaWAN-GW discard the messages.

**Reply by StevenCellist** [reply]
@drucknase the RadioLib stack is only fully tested on LW 1.1.
At some point I accidentally configured it for 1.0.2 and while it did work for me, it was not able to handle all MAC commands properly, which caused a downlink for every uplink which breaches the FUP very very quickly and is in general bad behaviour.
Maybe at some point someone will investigate 1.0.x, but I really really recommend 1.1.

But there should be no problem to have some nodes run 1.0.3 using LMIC and others running 1.1 from RadioLib - you don't need to deal with any of the differences. TTN will do that for you without you even noticing. So I recommend just setting it to 1.1!

**Reply by Dave0065** [reply]
The Dutch KPN lora network uses only the old version of Lora, the problem is that i have to insert a NWK_KEY but KPN require t left this variable empty. What could i try to make it joined?

**Reply by StevenCellist** [reply]
Please see the README in the examples/LoRaWAN folder :)

**Reply by Dave0065** [reply]
Thanks for the quick response! Now one step further, the KPN server gets the join request and sends the Join Accept message. But the join accept message does not result in the Heltec ESP32+Lora V3 module joining correctly. The status coming in on the serial monitor is -1116. In case it is useful, here is the link to the KPN manual of their gateway network.

https://docs.kpnthings.com/kpn-things/components/connectivity/lora

**Reply by StevenCellist** [reply]
While I do live in the Netherlands, I never spent the money on the KPN network. So please give us all possible information: logs from device (using protocol debug flag, see the troubleshooting guide) and logs/screenshots from the KPN dashboard. Please use a separate Discussion for that and we'll troubleshoot it :)
